src/snowflake/connector/ocsp_snowflake_async.py

```diff
@@ -0,0 +1,416 @@
+        ocsp_url = self.extract_ocsp_url(subject)
+        cert_id_enc = self.encode_cert_id_base64(self.decode_cert_id_key(cert_id))
+        if not ocsp_url:
+            telemetry_data.set_event_sub_type(OCSPTelemetryData.OCSP_URL_MISSING)
+            raise RevocationCheckError(
+                msg="No OCSP URL found in cert. Cannot perform Certificate Revocation check",
+                errno=ER_OCSP_URL_INFO_MISSING,
+            )
+        headers = {HTTP_HEADER_USER_AGENT: PYTHON_CONNECTOR_USER_AGENT}
+
+        if not OCSPServer.is_enabled_new_ocsp_endpoint():
+            actual_method = "post" if self._use_post_method else "get"
+            if self.OCSP_CACHE_SERVER.OCSP_RETRY_URL:
+                # no POST is supported for Retry URL at the moment.
+                actual_method = "get"
+
+            if actual_method == "get":
+                b64data = self.decode_ocsp_request_b64(ocsp_request)
+                target_url = self.OCSP_CACHE_SERVER.generate_get_url(ocsp_url, b64data)
+                payload = None
+            else:
+                target_url = ocsp_url
+                payload = self.decode_ocsp_request(ocsp_request)
+                headers["Content-Type"] = "application/ocsp-request"
+        else:
+            actual_method = "post"
+            target_url = self.OCSP_CACHE_SERVER.OCSP_RETRY_URL
+            ocsp_req_enc = self.decode_ocsp_request_b64(ocsp_request)
+
+            payload = json.dumps(
+                {
+                    "hostname": hostname,
+                    "ocsp_request": ocsp_req_enc,
+                    "cert_id": cert_id_enc,
+                    "ocsp_responder_url": ocsp_url,
+                }
+            )
+            headers["Content-Type"] = "application/json"
+
+        telemetry_data.set_ocsp_connection_method(actual_method)
+        if self.test_mode is not None:
+            logger.debug("WARNING - DRIVER IS CONFIGURED IN TESTMODE.")
+            test_ocsp_url = os.getenv("SF_TEST_OCSP_URL", None)
+            test_timeout = os.getenv(
+                "SF_TEST_CA_OCSP_RESPONDER_CONNECTION_TIMEOUT", None
+            )
+            if test_timeout is not None:
+                sf_timeout = int(test_timeout)
+            if test_ocsp_url is not None:
+                target_url = test_ocsp_url
+
+        self.debug_ocsp_failure_url = SnowflakeOCSP.create_ocsp_debug_info(
+            self, ocsp_request, ocsp_url
+        )
+        telemetry_data.set_ocsp_req(self.decode_ocsp_request_b64(ocsp_request))
+        telemetry_data.set_ocsp_url(ocsp_url)
+        telemetry_data.set_cert_id(cert_id_enc)
+
+        ret = None
+        logger.debug("url: %s", target_url)
+        sf_max_retry = SnowflakeOCSP.CA_OCSP_RESPONDER_MAX_RETRY_FO
+        if not self.is_enabled_fail_open():
+            sf_max_retry = SnowflakeOCSP.CA_OCSP_RESPONDER_MAX_RETRY_FC
+
+        async with make_client_session(asyncio.get_running_loop()) as session:
```

Comment:
I'm thinking we shall consolidate the usage of asyncio loop, I'm thinking of creating a loop in connection, and we pass it to the ocsp snowflake class making it a instance attribute, do you think it's possible?

Reply:
It is possible, but I don't think that we should intermingle connection class with the underlying network class.